packages/xsnap/src/build.js

```diff
@@ -85,119 +101,114 @@ function makeCLI(command, { spawn }) {
  * @param {{ git: ReturnType<typeof makeCLI> }} io
  */
 const makeSubmodule = (path, repoUrl, { git }) => {
-  /** @param {string} text */
-  const parseStatus = text =>
-    text
-      .split('\n')
-      // From `git submodule --help`:
-      // Show the status of the submodules. This will print the SHA-1 of the
-      // currently checked out commit for each submodule, along with the
-      // submodule path and the output of git describe for the SHA-1. Each
-      // SHA-1 will possibly be prefixed with - if the submodule is not
-      // initialized, + if the currently checked out submodule commit does
-      // not match the SHA-1 found in the index of the containing repository
-      // and U if the submodule has merge conflicts.
-      //
-      // We discovered that in other cases, the prefix is a single space.
-      .map(line => [line[0], ...line.slice(1).split(' ', 3)])
-      .map(([prefix, hash, statusPath, describe]) => ({
-        prefix,
-        hash,
-        path: statusPath,
-        describe,
-      }));
-
   return freeze({
     path,
     clone: async () => git.run(['clone', repoUrl, path]),
     /** @param {string} commitHash */
     checkout: async commitHash =>
       git.run(['checkout', commitHash], { cwd: path }),
     init: async () => git.run(['submodule', 'update', '--init', '--checkout']),
-    status: async () =>
-      git.pipe(['submodule', 'status', path]).then(parseStatus),
-    /** @param {string} leaf */
+    status: async () => {
+      const line = await git.pipe(['submodule', 'status', path]);
+      // From `git submodule --help`:
+      // status [--cached] [--recursive] [--] [<path>...]
+      //     Show the status of the submodules. This will print the SHA-1 of the
+      //     currently checked out commit for each submodule, along with the
+      //     submodule path and the output of git describe for the SHA-1. Each
+      //     SHA-1 will possibly be prefixed with - if the submodule is not
+      //     initialized, + if the currently checked out submodule commit does
+      //     not match the SHA-1 found in the index of the containing repository
+      //     and U if the submodule has merge conflicts.
+      //
+      // We discovered that in other cases, the prefix is a single space.
```

Comment:
This looks like a `fix`.

Reply:
Nope, just a move (it was previously `parseStatus` in `makeSubmodule`). There is a notional difference in the `describe` component, but a) it doesn't actually come up in actual `git status` output, and b) we don't look for it anyway.